### DISCUSSION TITLE
Quantization with monai networks

### DISCUSSION THREAD

**Discussion by o-fares** [Project-MONAI/MONAI#6117] [Ideas]
Hello, 

I am currently working on improving the size of a model obtained with a training on data with a MONAI UNet network.
One option to reduce its size and improve inference time is to use [quantization](https://pytorch.org/docs/stable/quantization.html), as it is available with the newest version of Pytorch.

To use quantization, there is a need to add quantized layers to the existing network. 
It is possible to do it by hand, by recreating a UNet network from scratch, or by following the structure of the Monai networks and add these new layers, but it would be better if it was already implemented.

My question is: Are you planning to add a quantization option to the existing networks in Monai ? 

Thank you.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Could you please help share some comments on this question?

Thanks in advance.

**Comment by ericspod** [comment]
We don't have any plans for quantisation yet but I'm sure this is a feature we'll want to look at once it becomes a little more mature. Unfortunately for now it does mean you'll have to implement the changes for yourself to use this, but if it's a proven useful feature I imagine we'll be supporting it.
